java/src/test/java/org/rocksdb/SequenceNumberAllocationTest.java

```diff
@@ -0,0 +1,198 @@
+        }
+
+        executor.shutdownNow();
+        assertTrue("Executor did not shutdown", executor.isShutdown());
+
+        try (RocksDB db = RocksDB.openReadOnly(options, dbFolder.getRoot().getAbsolutePath())) {
+            assertEquals("Lastest sequence number is not the one expected", sequenceNumberAfterFlush.get(), db.getLatestSequenceNumber());
+        } catch (RocksDBException e) {
+            fail("Could not reopen the database:" + e);
+        }
+    }
+
+    @Test
+    public void testSeqnoDataWrittenAfterFlush(WriteOptions writeOptions)
+            throws InterruptedException
+    {
+        ExecutorService executor = Executors.newFixedThreadPool(1);
+        AtomicLong sequenceNumberAfterFlush = new AtomicLong(0);
+        AtomicBoolean cancellable = new AtomicBoolean(false);
+        Options options = new Options().setCreateIfMissing(true);
+
+        Future<?> task = executor.submit(() -> {
+            try {
+                RocksDB db = RocksDB.open(options, dbFolder.getRoot().getAbsolutePath());
+                int count = fillDatabase(db, writeOptions, 0, 10);
+                assertEquals("Expected 10 values in the database", 10, count);
+                db.flush(new FlushOptions());
+                sequenceNumberAfterFlush.set(db.getLatestSequenceNumber());
+                assertEquals("Sequence number after flush should be 10", 10, sequenceNumberAfterFlush.get());
+                count = fillDatabase(db, writeOptions, 10, 20);
+                assertEquals("Expected 20 values in the database", 20, count);
+                assertEquals("Latest DB sequence number should be 20", 20, db.getLatestSequenceNumber());
+                cancellable.set(true);
+                while(true) {
+                    // Wait indefinitely
+                }
+            }
+            catch (RocksDBException e) {
+                fail("Encountered exception:" + e);
+            }
+        });
+
+        // Enough time to finish performing database operations
+        sleep(5 * 1000);
+
+        if(cancellable.get()) {
+            task.cancel(true);
+            assertTrue("Failed to cancel task", task.isCancelled());
+        } else {
+            fail("Task is not cancellable");
+        }
+
+        executor.shutdownNow();
+        assertTrue("Executor did not shutdown", executor.isShutdown());
+
+        int expectedSequenceNumber = writeOptions.disableWAL() ? 10 : 20;
+        try (RocksDB db = RocksDB.openReadOnly(options, dbFolder.getRoot().getAbsolutePath())) {
+            assertEquals("Lastest sequence number is not the one expected", expectedSequenceNumber, db.getLatestSequenceNumber());
+        } catch (RocksDBException e) {
+            fail("Could not reopen the database:" + e);
+        }
+    }
+
+    @Parameters(name = "{0}")
+    public static Iterable<Object> writeOptionsProvider() {
+        return Arrays.asList(
+                new WriteOptions().setDisableWAL(false),
```

Comment:
These WriteOptions objects need to be closed when you are finished with them. Instead consider using just a boolean parameter here. Also parameters for tests are best defined at the top of the tests as a static member variable rather than a function.